app/controllers/amr_heatmap_controller.rb

```diff
@@ -58,4 +63,174 @@ def amr_counts
+    end
+
+    return search_result
+  end
+
+  def gather_ontology_information(matching_entry, xml_dom)
+    ontology_information = {}
+    # Gathering properties
+    ontology_information["label"] = matching_entry.at_xpath("./rdfs:label").content
+    ontology_information["accession"] = matching_entry.at_xpath("./oboInOwl:id").content.split(':')[1]
+    ontology_information["description"] = matching_entry.at_xpath("./obo:IAO_0000115").content
+
+    # These properties are referred to by accession number in the matching OWL Class,
+    # but their labels are located elsewhere in the document.
+    remote_properties = {
+      "drugClass" => get_drug_classes(ontology_information["accession"], xml_dom),
+      "geneFamily" => matching_entry.xpath("./rdfs:subClassOf[not(*)]")
+    }
+
+    # Some drug resistances are only listed in the gene family entry
+    if remote_properties["drugClass"].empty?
+      drug_classes = []
+      remote_properties["geneFamily"].each do |node|
+        gene_family_accession = node.get_attribute("rdf:resource").split('_')[1]
+        drug_classes.concat(get_drug_classes(gene_family_accession, xml_dom))
+      end
+      remote_properties["drugClass"] = drug_classes
+    end
+
+    local_properties = {
+      "synonyms" => matching_entry.xpath("./oboInOwl:hasExactSynonym"),
+      "publications" => matching_entry.at_xpath("./following-sibling::owl:Axiom").xpath("./oboInOwl:hasDbXref")
+    }
+
+    local_properties.each do |ontology_key, property_nodes|
+      collection = []
+      property_nodes.each do |node|
+        collection.push(node.content)
+      end
+      ontology_information[ontology_key] = collection
+    end
+
+    remote_properties.each do |ontology_key, property_nodes|
+      collection = []
+      property_nodes.each do |node|
+        accession = node.get_attribute("rdf:resource").split('_')[1]
+        accession_class = xml_dom.at_xpath(".//owl:Class[@rdf:about='http://purl.obolibrary.org/obo/ARO_#{accession}']")
+        label = accession_class.at_xpath("./rdfs:label").content
+        description = accession_class.at_xpath("./obo:IAO_0000115").content
+        entry = { "label" => label, "description" => description }
+        collection.push(entry)
+      end
+      ontology_information[ontology_key] = collection
+    end
+    return ontology_information
+  end
+
+  def get_drug_classes(accession, xml_dom)
+    accession_class = xml_dom.at_xpath(".//owl:Class[@rdf:about='http://purl.obolibrary.org/obo/ARO_#{accession}']")
+    drug_class_node_markers = accession_class.xpath("./rdfs:subClassOf/owl:Restriction/owl:onProperty[@rdf:resource='http://purl.obolibrary.org/obo/RO#_confers_resistance_to_drug']")
+    drug_family_node_markers = accession_class.xpath("./rdfs:subClassOf/owl:Restriction/owl:onProperty[@rdf:resource='http://purl.obolibrary.org/obo/RO#_confers_resistance_to']")
+    drug_class_nodes = []
+    drug_class_node_markers.each do |node_marker|
```

Comment:
you could dedup here by doing `[drug_class_node_markers + drug_family_node_markers].each do ...`

Reply:
Deduped in a different manner